### DISCUSSION TITLE
Currency Error

### DISCUSSION THREAD

**Discussion by Dilosen** [ghostfolio/ghostfolio#3518] [Q&A] [answered]
When I attempt to import a list of transactions from the Johannesburg stock exchange, the format in Yahoo recognises the ticker.

The account is in ZAR (South African Rand) however Yahoo records in ZAc (South African cents). This mismatch is being flagged: 
activities.0.currency ("ZAR") does not match with currency of AIL.JO ("ZAc")

I first tried to change the currency column to ZAc or ZAC and neither recognised on the platform (although ZAc is added in Admin panel). The error then relates to ISO 4217, which means that ZAR is required, and ZAc isn't recognised. 

I then tried to change the account currency to ZAc but it does not save the change. When "Save" is pressed, the currency input field signals red, but there is no error message. I cannot add a new account using ZAc either. 

Not sure how to add these positions to the platform.

**Comment by Dilosen** [comment]
Added to those currencies, BTC does not seem to also want to be created or assigned to an account. 

The account works fine with AUD, JPY, USD

**Comment by dtslvr** [comment]
> The error then relates to ISO 4217, which means that ZAR is required, and ZAc isn't recognised.

> I cannot add a new account using ZAc either.

These issues will be fixed via https://github.com/ghostfolio/ghostfolio/pull/3529.

**Reply by Dilosen** [reply]
Thanks, looking forward to adding those currencies.

**Reply by Dilosen** [reply]
> > The error then relates to ISO 4217, which means that ZAR is required, and ZAc isn't recognised.
> 
> > I cannot add a new account using ZAc either.
> 
> These issues will be fixed via #3529.

Hi, I've updated to 2.92.0 but am still receiving errors for currencies: "activities.0.BTC must be a valid ISO4217 currency code"

I'm sure this tracked currencies, especially crypto? Am I doing something wrong? My data source is Yahoo. 

Edit: I am able to add the ZAc currencies.

**Reply by dtslvr** [reply]
> activities.0.BTC must be a valid ISO4217 currency code

This is expected behavior, you cannot use `BTC` for the `currency` field.

With this test file, I'm able to import an activity of an asset listed in `ZAc`: https://github.com/ghostfolio/ghostfolio/blob/main/test/import/ok-derived-currency.json

**Reply by Dilosen** [reply]
Pardon my understanding, but how do we add crypto tickers to the portfolio? I couldn't find any documentation on this. I assumed it was the ticker BTC in the currency field, albiet changing the datasource. 

I am able to add traditional FIAT currencies and it's working fine

**Reply by dtslvr** [reply, marked as answer]
> Pardon my understanding, but how do we add crypto tickers to the portfolio?

You can add cryptocurrency activities to your portfolio via the GUI. Please search for the currency by name, such as "Bitcoin", and you will find the ticker you want to add.

**Reply by Dilosen** [reply]
I have done this in the admin control. here is the job for BTCUSD retrieval. 

![tyu56](https://github.com/ghostfolio/ghostfolio/assets/16508683/f237abca-4a3b-4887-a0fa-108585d9d1ee)

I've added ETH to test; adding the currency works as expected, however here is the stack trace for the Job:

"Error: Error: Could not get historical market data for USDBTC (YAHOO) from 2024-06-23 to 2024-06-30: [HTTPError] Not Found\n    at DataGatheringProcessor.gatherHistoricalMarketData (/ghostfolio/apps/api/main.js:1:347991)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)",

"[HTTPError] Not Found\n" - my assumption is that the datasource is not set correctly for the cryptocurrencies. In the admin area, both ETH and BTC are 1:1 to USD. Other currencies pull fine.

![asdwqdsf](https://github.com/ghostfolio/ghostfolio/assets/16508683/5f85d99f-4992-46a2-a0a0-a68ed26a9946)

**Reply by dtslvr** [reply]
You can safely delete the entries `BTC` and `ETH` from _Exchange Rates_.

**Reply by Dilosen** [reply]
> > Pardon my understanding, but how do we add crypto tickers to the portfolio?
> 
> You can add cryptocurrency activities to your portfolio via the GUI. Please search for the currency by name, such as "Bitcoin", and you will find the ticker you want to add.

Is there a way to import this data via CSV? Manual implementation seems like a chore.

**Reply by dtslvr** [reply]
Yes, that is possible. [Here](https://github.com/ghostfolio/ghostfolio/blob/main/test/import/ok.csv) you will find a sample csv file in the expected format.
To retrieve the individual values (e.g. `Code`, `DataSource`, etc.) of an activity, you could export an existing one.

**Reply by Dilosen** [reply]
Perfect, I did an import yesterday with some success after I made some test data and exported the Json. 

The currency, for example Bitcoin, is pegged to the USD. In fact, all of the crypto's are pegged that way. Is there any way to change this? For example, a BTC/ZAR price pull? 

Otherwise, this issue is resolved.

**Reply by dtslvr** [reply]
> The currency, for example Bitcoin, is pegged to the USD. In fact, all of the crypto's are pegged that way.

Please see here: https://github.com/ghostfolio/ghostfolio/discussions/3344
